app/components/browse/paths.rb

```diff
@@ -9,6 +9,10 @@ def browse_path(args)
     end
   end
 
+  def browse_url
+    send("browse_#{action}_url")
+  end
```

Comment:
There's a hidden bug here, but we might not run into it. `browse_url` and `browse_path` are actually methods that are defined via `ActionView::Helpers::UrlHelper` where a route defined in `routes.rb` gets the appropriate path and url methods. Currently:

```
 get '/browse', controller: :browse, action: :index 
```

Gets translated to `browse_path` and `browse_url`. What we're doing in this path module is overriding that. What would happen in the above example is that, if we were using `/browse` this would call `browse_index_url` which doesn't exist and would raise NoMethodError. However, I don't think we're using browse_url anywhere in the way it was originally intended. Still, we might want to guard ourselves against this.

Another option is to create a different method other than `browse_path` or `browse_url` and re-route accordingly.

Reply:
Ahh gotcha. That makes sense. Would it be sufficient to just check for the index action like you did for `browse_path` above?

```
def browse_url
  if action == 'index'
    super(args)
  else  
    send("browse_#{action}_url")
  end
end
```

Reply:
yeah, I think so. That's what we're doing with `browse_path`. There's probably a more elegant way once we get a named path for call numbers.